src/seedsigner/views/screensaver.py

```diff
@@ -183,47 +194,76 @@ def start(self):
+                    
+                    # Calculate pixel movement based on elapsed time and target speed
+                    # Accumulate fractional movements to avoid getting stuck at zero
+                    accumulated_x += elapsed_time * current_speed * self.vector_x
+                    accumulated_y += elapsed_time * current_speed * self.vector_y
+                    
+                    # Only move when we've accumulated at least 1 pixel of movement
+                    pixels_to_move_x = int(accumulated_x)
+                    pixels_to_move_y = int(accumulated_y)
+                    
+                    # Remove the used whole pixels from accumulator
+                    accumulated_x -= pixels_to_move_x
+                    accumulated_y -= pixels_to_move_y
+                    
+                    # Apply movement
+                    if pixels_to_move_x != 0 or pixels_to_move_y != 0:
+                        self.cur_x += pixels_to_move_x
+                        self.cur_y += pixels_to_move_y
+
                     # Must crop the image to the exact display size
                     crop = self.image.crop((
                         self.cur_x, self.cur_y,
                         self.cur_x + self.renderer.canvas_width, self.cur_y + self.renderer.canvas_height))
                     self.renderer.disp.ShowImage(crop, 0, 0)
 
-                    self.cur_x += self.increment_x
-                    self.cur_y += self.increment_y
-
-                    # At each edge bump, calculate a new random rate of change for that axis
-                    if self.cur_x < self.min_coords[0]:
-                        self.cur_x = self.min_coords[0]
-                        self.increment_x = self.rand_increment()
-                        if self.increment_x < 0.0:
-                            self.increment_x *= -1.0
-                    elif self.cur_x > self.max_coords[0]:
-                        self.cur_x = self.max_coords[0]
-                        self.increment_x = self.rand_increment()
-                        if self.increment_x > 0.0:
-                            self.increment_x *= -1.0
-
-                    if self.cur_y < self.min_coords[1]:
-                        self.cur_y = self.min_coords[1]
-                        self.increment_y = self.rand_increment()
-                        if self.increment_y < 0.0:
-                            self.increment_y *= -1.0
-                    elif self.cur_y > self.max_coords[1]:
-                        self.cur_y = self.max_coords[1]
-                        self.increment_y = self.rand_increment()
-                        if self.increment_y > 0.0:
-                            self.increment_y *= -1.0
+                    # Handle boundary collisions and ensure we don't get stuck
+                    if self.cur_x <= self.min_coords[0]:
+                        self.cur_x = self.min_coords[0] + 1  # Move slightly away from edge
```

Comment:
No need to worry about 1 pixel differences. The logo bounce is calculated from the center of the logo; there aren't any extreme edge scenarios to worry about.